### ISSUE TITLE
Group Configurations in fieldset where possible

### ISSUE THREAD

**Issue by dhinrichs-scottlogic** [ScottLogic/prompt-injection#677]
# Feature Request

## Description
Wrap the slider configurations (e.g. Model Temperature header, slider & info button) in a fieldset so that they are programmatically linked. Remove any duplicate labels that are not needed anymore. 
Style the element so that there is no visual difference compared to before (the fieldset element will automatically add a border, that can be removed). 

This might be possible with other configurations as well, but with the slider configs it should pretty straightforward. 

## Screenshots
Fieldset adds a border, remove this so that it looks as it did before
![Image](https://github.com/ScottLogic/prompt-injection/assets/125262707/c24e2610-b1f7-4f55-91ba-7b99dfdcd2a1)

## Additional context
Accessibility improvement to reduce noise in the accessibility tree and correctly link related inputs together

## Acceptance criteria
**GIVEN** a user navigates the configurations with assistive tech
**WHEN** they move through one section of the configurations
**THEN** the interactive elements are grouped and labels aren't repeated unnecessarily

**Comment by pmarsh-scottlogic** [comment]
## Testing with NVDA:
### Forward tabbing through the model configurations
![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/148a2e12-28d1-4675-a703-7c1eb1ca3fbc)
"Top P grouping more info button"

![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/d338e2dd-6bac-4d8e-8a9f-12b2267ff68f)
"Top P slider 1"

### Reverse tabbing
![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/d338e2dd-6bac-4d8e-8a9f-12b2267ff68f)
"Top P grouping, Top P slider 1" [repetition]

![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/148a2e12-28d1-4675-a703-7c1eb1ca3fbc)
"More info button"

So it seems like it reads out the grouping name when you first focus an element in the grouping, but it doesn't read the grouping name again. 
Now, because we have the configuration name (e.g. "Top P") included in the label for the sliders, but not the info buttons, this causes some inconsistencies when you tab forwards and backwards. Possibly this is confusing. But maybe it's fine. Personally, I'd suggest removing the configuration name from the slider too. Keen to hear others' thoughts!

Note: I've only tested with NVDA. Other screen readers might behave differently.

**Comment by dhinrichs-scottlogic** [comment]
For NVDA I think it's fine, but now it's not working very well for Narrator... 
Let's have a chat when you're back @pmarsh-scottlogic, this needs some more consideration and rubber ducking

**Comment by pmarsh-scottlogic** [comment]
Had a discussion with @dhinrichs-scottlogic  and happy to leave as is :)
